File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        vowels = "aeiou"
        cnt = ans = sum(s[i] in vowels for i in range(k))

        for i in range(k, len(s)):
            cnt += (s[i] in vowels) - (s[i - k] in vowels)
            ans = max(ans, cnt)

        return ans
                

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.go
func maxVowels(s string, k int) int {
    ans := 0    
    for i := 0; i < k; i++ {
        if isVowel(s[i]) {
            ans += 1
        }
    }

    cur := ans
    for i := k; i < len(s); i++ {
        if isVowel(s[i]) {
            cur += 1
        }

        if isVowel(s[i-k]) {
            cur -= 1
        }

        ans = max(ans, cur)
    }

    return ans
}



func isVowel(c byte) bool {
    vowels := []byte{'a', 'e', 'i', 'o', 'u'}
    for _, v := range vowels {
        if v == c {
            return true
        }
    }

    return false
}
